### ISSUE TITLE
appsettingscontroller warnings in app settings page

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#17433]
### Steps
1. Download git version of the calendar app, put it into "apps"
2. Enable "calendar" app in the apps page
3. Refresh the apps page

### Expected result
No warnings in log

### Actual result
```
{"reqId":"xpqNhhF8H+G4mYblJnqN","remoteAddr":"127.0.0.1","app":"PHP","message":"array_key_exists() expects parameter 2 to be array, null given at \/srv\/www\/htdocs\/owncloud\/settings\/controller\/appsettingscontroller.php#176","level":3,"time":"2015-07-07T10:04:42+00:00","method":"GET","url":"\/owncloud\/index.php\/settings\/apps\/list?category=0&includeUpdateInfo=0"}
{"reqId":"wr5za4UI3gA1RYHTFGqu","remoteAddr":"127.0.0.1","app":"PHP","message":"array_key_exists() expects parameter 2 to be array, null given at \/srv\/www\/htdocs\/owncloud\/settings\/controller\/appsettingscontroller.php#176","level":3,"time":"2015-07-07T10:04:43+00:00","method":"GET","url":"\/owncloud\/index.php\/settings\/apps\/list?category=0&includeUpdateInfo=1"}
```

**Comment by PVince81** [comment]
### Version
Observed on ownCloud 8.2 pre-alpha (git master 645600ae03101c1212576022548781641f2b4970)

**Comment by tony5** [comment]
I have notice this error on a fresh install of OC 8.1 

array_key_exists() expects parameter 2 to be array, null given at /var/www/owncloud/settings/controller/appsettingscontroller.php#176

**Comment by CommanderAlchemy** [comment]
I have this on OC 8.1

Trying to enable Documents app.
Error	PHP	array_key_exists() expects parameter 2 to be array, null given at /usr/share/webapps/owncloud/settings/controller/appsettingscontroller.php#192

**Comment by PVince81** [comment]
I saw it happen once too, trying to enable the documents app.
So I started over but then it didn't appear any more.
Very strange...

**Comment by PVince81** [comment]
Must be a heisenbug... try and observe it then it disappears.

The code in question is here https://github.com/owncloud/core/blob/stable8.1/settings/controller/appsettingscontroller.php#L192

From reading the code, the warning would appear if `$remoteAppEntry` is null, which means the remote app was NOT found on the app store server.
And this is only for apps that have OCSID set but no "level" attribute. It seems if the level isn't defined locally it tries to retrieve it from the server.

Maybe the server fails randomly ?

**Comment by PVince81** [comment]
Aha of course: I opened the apps page but had "appstoreenabled => false" in my config.

### Steps to reproduce
1. Setup ownCloud
2. Install "calendar" app from git
3. Disable app store by setting "appstoreenabled" to false in config.php
4. Go to the apps page and select "Not enabled".

In this case it will try and retrieve the calendar app's info from the app store, but since it's disabled, `null` is returned.

This guy fixed it: https://github.com/owncloud/core/pull/17628

**Comment by CommanderAlchemy** [comment]
@PVince81 I Probably have the same setting, but as a workaround would be to set it true?

**Comment by PVince81** [comment]
Yes, if you like.

**Comment by CommanderAlchemy** [comment]
@PVince81, seems it fixed the error, removed the app from /apps manually and then enabled it from the owncloud apps it worked and no error but does not appear in apps and no errors about it odd.

**Comment by PVince81** [comment]
Fixed through another PR: https://github.com/owncloud/core/pull/17606

**Comment by MorrisJobke** [comment]
Also a duplicate of #17587

**Comment by peterkahl** [comment]
I have the exact same problem.
```
array_key_exists() expects parameter 2 to be array, null given at /srv/www/sub.domain.tld/public_html/settings/controller/appsettingscontroller.php#176
```
I am using version 8.1.0.
The above error is most likely related to cURL timing out. (I am denying the connection in my firewall.)

**Comment by MorrisJobke** [comment]
> I am using version 8.1.0.

@peterkahl This is most likely fixed in #17606 which is included in 8.1.1 and 8.2.0. Please give them a try once they are released.
